src/sentry/integrations/repository/issue_alert.py

```diff
@@ -140,3 +151,32 @@ def create_notification_message(
                 extra=data.__dict__,
             )
             raise
+
+    def get_all_parent_notification_messages_by_filters(
+        self, group_ids: list[int] | None = None, project_ids: list[int] | None = None
+    ) -> Generator[IssueAlertNotificationMessage, None, None]:
+        """
+        If no filters are passed, then all parent notification objects are returned.
+        Because all parent notification objects can be returned, this method leverages generator to control the usage of
+        memory in the application.
+        It is up to the caller to iterate over all the data, or store in memory if they need all objects concurrently.
+        """
+        try:
+            group_id_filter = Q(rule_fire_history__group__id__in=group_ids) if group_ids else Q()
+            project_id_filter = (
+                Q(rule_fire_history__project_id__in=project_ids) if project_ids else Q()
+            )
+
+            query = self._model.objects.filter(group_id_filter & project_id_filter).filter(
+                self._parent_notification_message_base_filter()
+            )
+        except Exception as e:
```

Comment:
What kind of exception do we expect here?

Reply:
oh haha, good point! Because this doesn't actually query anything yet, I should move this to around the for loop

Reply:
done, moved it to the for loop!